test/integration/puppetlabs/services/jruby/jruby_pool_int_test.clj

```diff
@@ -55,9 +56,25 @@
 (def script-to-check-if-constant-is-defined
   "! $instance_id.nil?")
 
+(defn await-jruby-agents
+  "Wait for pending actions on agents associated with the jruby pool
+  service to free up."
+  [{:keys [pool-agent flush-instance-agent]}]
+  ;; Dump a dummy action into each of the jruby pool agents and await that
+  ;; action's completion.  This assumes that any previously existing
+  ;; actions would have been flushed through prior to the new one being
+  ;; processed.
+  (jruby-agents/send-agent pool-agent #(constantly true))
```

Comment:
Yeah, re-reading that comment now, I think you're right.  I had originally read that as `await` only blocking until actions that had originated from the calling thread - as opposed to any thread - had completed.  But when I tried this out locally, I see that even the flush actions originating from a webserver-controlled thread are flushed through from an `await` call on the main test thread.  Given that, I took out the dummy actions in 9ca5cd1.  Since the tests only need to ensure that the pool agent is flushed before registering a wait for flush completion and for the flush instance agent to be flushed through before shutdown in the one test that needs it, I just removed this helper function and replaced with the appropriate in-line awaits.